worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/BuiltInClipboardFormat.java

```diff
@@ -86,56 +86,65 @@ public boolean isFormat(File file) {
 
     },
 
-    /**
-     * The Schematic format used by MCEdit.
-     */
-    MCEDIT_SCHEMATIC("mcedit", "mce", "schematic") {
+    SPONGE_SCHEMATIC("sponge", "schem") {
 
         @Override
         public String getPrimaryFileExtension() {
-            return "schematic";
+            return "schem";
         }
-
         @Override
         public ClipboardReader getReader(InputStream inputStream) throws IOException {
-            if (inputStream instanceof FileInputStream) {
-                inputStream = new ResettableFileInputStream((FileInputStream) inputStream);
-            }
-            BufferedInputStream buffered = new BufferedInputStream(inputStream);
-            NBTInputStream nbtStream = new NBTInputStream(new BufferedInputStream(new GZIPInputStream(buffered)));
-            SchematicReader input = new SchematicReader(nbtStream);
-            input.setUnderlyingStream(inputStream);
-            return input;
+            NBTInputStream nbtStream = new NBTInputStream(new GZIPInputStream(inputStream));
+            return new SpongeSchematicReader(nbtStream);
         }
 
         @Override
         public ClipboardWriter getWriter(OutputStream outputStream) throws IOException {
-            throw new IOException("This format does not support saving, use `schem` or `sponge` as format"); // Is more helpful
+            NBTOutputStream nbtStream = new NBTOutputStream(new GZIPOutputStream(outputStream));
+            return new SpongeSchematicWriter(nbtStream);
         }
 
         @Override
         public boolean isFormat(File file) {
-            String name = file.getName().toLowerCase(Locale.ROOT);
-            return name.endsWith(".schematic") || name.endsWith(".mcedit") || name.endsWith(".mce");
+            try (NBTInputStream str = new NBTInputStream(new GZIPInputStream(new FileInputStream(file)))) {
+                NamedTag rootTag = str.readNamedTag();
+                if (!rootTag.getName().equals("Schematic")) {
+                    return false;
+                }
+                CompoundTag schematicTag = (CompoundTag) rootTag.getTag();
+
+                // Check
+                Map<String, Tag> schematic = schematicTag.getValue();
+                if (!schematic.containsKey("Version")) {
+                    return false;
+                }
+            } catch (Exception e) {
+                return false;
+            }
+
+            return true;
         }
```

Comment:
Jesse once yelled at me when I tried to change this because he claimed that checking the file extension was much faster than checking for a schematic tag and version key in a file. How long does this operation usually take before we make this change?

Reply:
By re-arranging to the MCEDIT format last in the enum this won't matter. Trying to minimise the diff as FAWE doesn't change behaviour. Altering things for the sake of altering them is dumb.